Add SSOTeamID to Team struct

xlgmokha opened the pull request:
<!--
Thank you for contributing to hashicorp/go-tfe!

Here's what to expect after the pull request is opened:

The test suite contains many acceptance tests that are run against a test version of Terraform Cloud and additional testing is done against Terraform Enterprise. You can read more about running the tests against your own Terraform Enterprise environment in TESTS.md. Our CI system (Circle) will not test your fork unless you are an authorized employee, so a HashiCorp maintainer will initiate the tests or you and report any missing tests or simple problems. In order to speed up this process, it's not uncommon for your commits to be incorportated into another PR that we can commit test changes to.

Your change, depending on its impact, may be released in the following ways:

  1. For impactful bug fixes, it can be released fairly quickly as a patch release
  2. For noncritical bug fixes and new features, it will be incorporated into the next minor version release
  3. For breaking changes (those changes that alter the public method signatures), more consideration must be made and alternatives may be considered, depending on upgrade difficulty and release schedule.

Please fill out the remaining template to assist code reviewers and testers with incorporating your change. If a section does not apply, feel free to delete it.
-->

## Description

<!-- Describe why you're making this change. -->

The [teams API](https://www.terraform.io/cloud-docs/api-docs/teams) is being
updated to include a new attribute called `sso-team-id` to allow SSO enabled
organizations to use unique identifiers from external systems to control team
membership.

## Testing plan

TBD
<!--
1.  _Describe how to replicate_
1.  _the conditions under which your code performs its purpose,_
1.  _including example code to run where necessary._
-->

<!--
## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- [API documentation](https://www.terraform.io/docs/cloud/api/xxxx.html)
- [Related PR](https://github.com/terraform-providers/terraform-provider-tfe/pull/xxxx)

-->

## Output from tests (HashiCorp employees only)

<!--
_Please run the tests locally for any files you changes and include the output here._
$ TFE_ADDRESS="https://example" TFE_TOKEN="example" TF_ACC="1" go test ./... -v -tags=integration -run TestFunctionsAffectedByChange

...
```
-->
```

```bash
モ TFE_TOKEN=$TOKEN TFE_ADDRESS=https://tfcdev-82d111f9.ngrok.io TF_ACC=1 go test ./... -run TestFunctionsAffectedByChange
ok      github.com/hashicorp/go-tfe     0.006s [no tests to run]
?       github.com/hashicorp/go-tfe/examples/organizations      [no test files]
?       github.com/hashicorp/go-tfe/examples/workspaces [no test files]
?       github.com/hashicorp/go-tfe/mocks       [no test files]
```

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/go-tfe?pullRequest=364)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/go-tfe?pullRequest=364)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/go-tfe?pullRequest=364).</sub>

xlgmokha commented on team_integration_test.go:
```diff
@@ -121,15 +137,16 @@ func TestTeamsRead(t *testing.T) {
 		},
 	}
-	tm, err := client.Teams.Create(ctx, orgTest.Name, opts)
+	ssoTeam, err := client.Teams.Create(ctx, orgTest.Name, opts)
```
`tm` was being overwritten [down below](https://github.com/hashicorp/go-tfe/blob/e14f5437536ac9ff37268acbc3f41c12f4e7e21e/team_integration_test.go#L137).

sebasslash reviewed (changes requested):
Changes look good! Some minor things mentioned below...

Another thing to note is to squash the `test:` commits for a cleaner commit history!

xlgmokha commented:
I squashed the commits and added `omitempty` where I think it's appropriate.

```bash
モ ENABLE_BETA=1 TFE_ADDRESS=https://... TF_ACC=1 go test -v ./... -tags=integration -run TestTeams
=== RUN   TestTeamsList
=== RUN   TestTeamsList/without_list_options
    team_integration_test.go:37: paging not supported yet in API
=== RUN   TestTeamsList/with_list_options
    team_integration_test.go:43: paging not supported yet in API
=== RUN   TestTeamsList/without_a_valid_organization
--- PASS: TestTeamsList (2.13s)
    --- SKIP: TestTeamsList/without_list_options (0.25s)
    --- SKIP: TestTeamsList/with_list_options (0.00s)
    --- PASS: TestTeamsList/without_a_valid_organization (0.00s)
=== RUN   TestTeamsCreate
=== RUN   TestTeamsCreate/with_valid_options
=== RUN   TestTeamsCreate/with_sso-team-id
=== RUN   TestTeamsCreate/when_options_is_missing_name
=== RUN   TestTeamsCreate/when_options_has_an_invalid_organization
--- PASS: TestTeamsCreate (1.48s)
    --- PASS: TestTeamsCreate/with_valid_options (0.40s)
    --- PASS: TestTeamsCreate/with_sso-team-id (0.23s)
    --- PASS: TestTeamsCreate/when_options_is_missing_name (0.00s)
    --- PASS: TestTeamsCreate/when_options_has_an_invalid_organization (0.00s)
=== RUN   TestTeamsRead
=== RUN   TestTeamsRead/when_the_team_exists
=== RUN   TestTeamsRead/when_the_team_exists/visibility_is_returned
=== RUN   TestTeamsRead/when_the_team_exists/permissions_are_properly_decoded
=== RUN   TestTeamsRead/when_the_team_exists/organization_access_is_properly_decoded
=== RUN   TestTeamsRead/when_the_team_exists/SSO_team_id_is_returned
=== RUN   TestTeamsRead/when_the_team_does_not_exist
=== RUN   TestTeamsRead/without_a_valid_team_ID
--- PASS: TestTeamsRead (1.96s)
    --- PASS: TestTeamsRead/when_the_team_exists (0.18s)
        --- PASS: TestTeamsRead/when_the_team_exists/visibility_is_returned (0.00s)
        --- PASS: TestTeamsRead/when_the_team_exists/permissions_are_properly_decoded (0.00s)
        --- PASS: TestTeamsRead/when_the_team_exists/organization_access_is_properly_decoded (0.00s)
        --- PASS: TestTeamsRead/when_the_team_exists/SSO_team_id_is_returned (0.00s)
    --- PASS: TestTeamsRead/when_the_team_does_not_exist (0.16s)
    --- PASS: TestTeamsRead/without_a_valid_team_ID (0.00s)
=== RUN   TestTeamsUpdate
=== RUN   TestTeamsUpdate/with_valid_options
=== RUN   TestTeamsUpdate/when_the_team_does_not_exist
=== RUN   TestTeamsUpdate/without_a_valid_team_ID
--- PASS: TestTeamsUpdate (1.72s)
    --- PASS: TestTeamsUpdate/with_valid_options (0.38s)
    --- PASS: TestTeamsUpdate/when_the_team_does_not_exist (0.18s)
    --- PASS: TestTeamsUpdate/without_a_valid_team_ID (0.00s)
=== RUN   TestTeamsDelete
=== RUN   TestTeamsDelete/with_valid_options
=== RUN   TestTeamsDelete/without_valid_team_ID
--- PASS: TestTeamsDelete (1.31s)
    --- PASS: TestTeamsDelete/with_valid_options (0.34s)
    --- PASS: TestTeamsDelete/without_valid_team_ID (0.00s)
PASS
ok      github.com/hashicorp/go-tfe     (cached)
?       github.com/hashicorp/go-tfe/examples/organizations      [no test files]
?       github.com/hashicorp/go-tfe/examples/workspaces [no test files]
?       github.com/hashicorp/go-tfe/mocks       [no test files]
```

sebasslash reviewed (approved):
We're not in Kansas anymore 🔥

xlgmokha commented on team_integration_test.go:
```diff
@@ -321,7 +346,7 @@ func TestTeamCreateOptions_Marshal(t *testing.T) {
 	require.NoError(t, err)
 
-	expectedBody := `{"data":{"type":"teams","attributes":{"name":"team name","organization-access":{"manage-policies":true},"visibility":"organization"}}}
+	expectedBody := `{"data":{"type":"teams","attributes":{"name":"team name","organization-access":{"manage-policies":true},"sso-team-id":null,"visibility":"organization"}}}
```
I think this change might cause failures when run from CI because the associated feature flag is disabled by default. Should I undo this change?

radditude replied:
We can bring this back once we remove the feature flag and officially un-beta-fy this field!

sebasslash replied:
You can use the `skipIfBeta(t)` helper for tests that require a feature sitting behind a feature flag! I think it still eludes me which feature flags are enabled for CI.

radditude reviewed (approved):
This looks great to me! Though I will defer to a maintainer for final approval 😆

sebasslash reviewed (approved):
_Not_ from a Jedi! 🔥 Good work
